design/crd-configstatus.md

```diff
@@ -0,0 +1,348 @@
+        return ctrl.Result{}, nil
+    }
+
+    syncResult = r.Handler(r.Logger, cfg.Pools)
+    return ctrl.Result{}, nil
+}
+```
+
+### Server-Side Apply for Concurrent Updates
+
+ConfigurationStatus uses [Kubernetes Server-Side
+Apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/) to
+handle concurrent status updates from multiple controllers without conflicts.
+
+Without server-side apply, concurrent updates cause **resource version conflicts**:
+```
+Operation cannot be fulfilled on configurationstatuses.metallb.io "config-status":
+the object has been modified; please apply your changes to the latest version and try again
+```
+
+Each controller becomes a unique **field manager** and owns its specific condition:
+- `controller/poolReconciler` owns the `controller/poolReconcilerValid` condition
+- `speaker-node1/configReconciler` owns the `speaker-node1/configReconcilerValid` condition
+- `speaker-node2/configReconciler` owns the `speaker-node2/configReconcilerValid` condition
+
+When a controller updates its condition:
+1. Creates a partial ConfigurationStatus object with only its condition
+2. Uses `client.Status().Patch()` with `client.Apply` patch type
+3. Specifies field owner via `client.FieldOwner(owner)`
+4. Kubernetes API server merges the condition into the existing conditions array
+5. No conflicts - each field manager owns different fields
+
+
+**Benefits:**
+- No resource version conflicts
+- Controllers can update concurrently
+- Clear field ownership tracking
+- Automatic conflict resolution
+- Kubernetes-native approach
+
+To see which field manager owns which fields:
+
+```bash
+kubectl get configurationstatus config-status -n metallb-system --show-managed-fields -o yaml
+```
+## Resource Lifecycle and Cleanup
+
+The ConfigurationStatus resource is **created at runtime** by the ConfigurationStatusReconciler, not included in
+installation manifests. This means:
+
+- Created automatically when the unique controller configstatus starts
+- Not deleted when running `kubectl delete -f metallb.yaml` or `helm uninstall metallb`
+- Requires manual cleanup: `kubectl delete configurationstatus config-status -n metallb-system`
+- Other components never fail if they cannot patch their condition
+- Must review stale conditions, are they removed?
+
+# FEEDBACK from Fede
+
+https://github.com/metallb/metallb/pull/2856#issuecomment-3388777537
+
+> I am not sure I like this approach, and this is not exactly what was drafted
+> in the related design proposal.
+
+We need to clarify the drafted proposal, questions on the other doc
+
+> What the proposal said was either to cut corners making one single global
+> configuration status managed by the controller (because the controller is
+> responsible of the majority of the configuration) OR a per speaker
+> configuration (which might make sense now with all the node selectors we have).
+
+This proposal does have "single global config status" AND each speaker adds each status.
```

Comment:
it's a single global object, which becomes messy if there are 1000 nodes in the cluster